src/theory/arith/arith_lemma.h

```diff
@@ -0,0 +1,76 @@
+/*********************                                                        */
+/*! \file arith_lemma.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Gereon Kremer
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief ArithLemma class, derived from Lemma.
+ **/
+
+#ifndef CVC4__THEORY__ARITH__ARITH_LEMMA_H
+#define CVC4__THEORY__ARITH__ARITH_LEMMA_H
+
+#include <tuple>
+#include <vector>
+
+#include "expr/node.h"
+#include "theory/arith/nl/inference.h"
+#include "theory/inference_manager_buffered.h"
+#include "theory/output_channel.h"
+
+namespace CVC4 {
+namespace theory {
+namespace arith {
+
+/**
+ * The data structure for a single lemma to process by the arithmetic theory,
```

Comment:
I'm a bit confused why this class is necessary? I was actually expecting that the existing NlLemma class would derive from Lemma.  That way these members (which are specific to non-linear) continues to live in nl/.  I think the current design allows this, right?

Reply:
As discussed offline: We now have `NlLemma` derive from `ArithLemma`.